### ISSUE TITLE
`CA2000` should report `System.Diagnostics.Process`es returned from static methods like `GetCurrentProcess()`

### ISSUE THREAD

**Issue by LukasGelke** [dotnet/roslyn-analyzers#6688]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000)

### Describe the improvement

Even e.g. [CA1837](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1837) essentially mentions that, "It's easy to forget to call `Dispose()` when using `Process.GetCurrentProcess().Id`".
Yes, CA1837 is useful. But sometimes that won't cut it, and one needs a `Process` for some thing other than Id Or Path/FileName.

Also, wouldn't it be "common courtesy" to use `Dispose()` (or `using`) on all objects that implement `IDisposable` anyway?

### Describe suggestions on how to achieve the rule

I'v looked at the code, but it's a but beyond my current knowledge of Roslyn.
But the Rule already detects the constructor, so why not the static methods?

### Additional context

```cs
using System;
using System.Diagnostics;

namespace MyIssue;

public sealed class Example
{
	public void Do()
	{
	  Process newProcess = new();
	  Process current = Process.GetCurrentProcess();
	  Process byId = Process.GetProcessById(1337);
	}
}
```
notice, how only `newProcess` gets reported:
![image](https://github.com/dotnet/roslyn-analyzers/assets/71026023/7d5fe520-e758-43a7-817d-aae38f66abee)

**Comment by mavasani** [comment]
The core issue is that we cannot treat all methods whose return type is disposable to be factory methods that are returning a new disposable object. This leads to a lot of false positives wherein the called method might actually be just returning a disposable object whose dispose ownership is still held with the type on which the method is defined. We have some heuristics to determine some common factory methods, but they cannot cover all cases.

There are couple of approaches that we can take here:
1. Short term - define a new editorconfig option that allows users to specify comma separated list of signatures for dispose factory methods and enhance the dispose analysis to account for these. We already have precedence for such options in the repo, so this shouldn't be too difficult to implement if someone wanted to submit a PR. For example, see https://github.com/dotnet/roslyn-analyzers/blob/main/docs/Analyzer%20Configuration.md#null-check-validation-methods and https://github.com/dotnet/roslyn-analyzers/commit/6539472700e2b31a1eefb9cf08f41da0d98e589b
2. Long term - we need https://github.com/dotnet/runtime/issues/29631 to be implemented to allow analyzers to correctly determine dispose ownership transfers.
